auth/src/main/java/com/github/twitch4j/auth/TwitchAuth.java

```diff
@@ -33,7 +33,7 @@ public TwitchAuth(CredentialManager credentialManager, String clientId, String c
 
     public static void registerIdentityProvider(CredentialManager credentialManager, String clientId, String clientSecret, String redirectUrl) {
         // register the twitch identityProvider
-        Optional<OAuth2IdentityProvider> ip = credentialManager.getOAuth2IdentityProviderByName("twitch");
+        Optional<TwitchIdentityProvider> ip = credentialManager.getIdentityProviderByName("twitch", TwitchIdentityProvider.class);
         if (!ip.isPresent()) {
```

Comment:
improbable breaking change here: if credential manager already has a `"twitch"` identity provider registered that isn't a `TwitchIdentityProvider`, then an exception will be thrown

Reply:
This change is fine imo, i think it is very unlikely that anyone did this.